exchange/bidder.go

```diff
@@ -280,27 +280,27 @@ func addNativeTypes(bid *openrtb2.Bid, request *openrtb2.BidRequest) (*nativeRes
 
 func setAssetTypes(asset nativeResponse.Asset, nativePayload nativeRequests.Request) error {
 	if asset.Img != nil {
-		if tempAsset, err := getAssetByID(asset.ID, nativePayload.Assets); err == nil {
+		if tempAsset, err := getAssetByID(*asset.ID, nativePayload.Assets); err == nil {
```

Comment:
The `asset.ID` now needs a `nil` check. Consider returning an error right after `asset.Img != nil` if the `asset.ID` is `nil`. To ensure the code behaves correctly, please add a test for when the id is omitted from the native json.